accepted/2021/default_string_globalization.md

```diff
@@ -0,0 +1,390 @@
+# Improving the _string_ developer experience
+
+Questions? Contact: @tarekgh @safern @GrabYourPitchforks
+
+## Problem summary
+
+APIs on `string` are not consistently ordinal or linguistic, and this leads to confusion among our developer community and bugs in applications. https://github.com/dotnet/runtime/issues/43956 describes this pain in more detail, and it aggregates issues where developers experienced breaks when upgrading from .NET Core 3.1 to .NET 5 due to runtime changes exposing latent bugs in their own applications.
+
+That GitHub issue describes various options we were considering, including relying solely on Roslyn analyzers or providing runtime compatibility switches. None of these issues is very palatable because they either force the developers to think about globalization concerns or they make it difficult for library authors to reason about what behaviors their components will exhibit at runtime.
+
+This draft proposes an alternate solution which attempts to provide the best of all worlds: steering new developers toward success, letting people with existing code bases know when they have issues, and allowing existing compiled DLLs to work largely as expected.
+
+## Characteristics of an ideal solution
+
+An ideal solution to this problem should have two qualities.
+
+1. `string` and related APIs should default to ordinal behavior unless the caller makes an explicit request for a different behavior.
+
+2. Developers should not be exposed to any globalization-related concepts unless they explicitly seek globalization support.
```

Comment:
Is this claim specific to APIs on String?  For example, how does this relate to formatting/parsing APIs on types like Int32, DateTime, etc., which all factor in the current culture by default?

Reply:
Perhaps I should clarify this. The concept of "ordinal" only applies to APIs which operate on or transform strings / sequences of chars. (Think: `ReadOnlySpan<char>` APIs in `MemoryExtensions`.)

For APIs which are about changing representation between string and non-string types (e.g., _string_ \<-\> _int_), I don't suggest any behavioral change. These APIs necessarily have to perform some kind of linguistic translation, so they may as well fall back to _current culture_ as they do today. That also helps solidify `Parse` and `ToString` as helpers for "please convert this data from / to a representation which is understandable by the current user."

Reply:
> The concept of "ordinal" only applies to APIs which operate on or transform strings / sequences of chars.

Yes.  I was commenting on the second goal/requirement: "Developers should not be exposed to any globalization-related concepts unless they explicitly seek globalization support."  There are many globalization-related concepts unrelated to ordinal.